learn/variables_and_types.md

```diff
@@ -0,0 +1,200 @@
+## Variable Definition
+
+What is a variable in Python? It is a named location in the computer's memory that stores a value. It is like a container that can hold different types of data, such as numbers, strings or booleans.
+
+To create a variable in Python, you need to give it a name and assign a value to it using the assignment operator `=`.
+
+Here's an example of a variable declaring in Python:
+
+```python
+# create a variable called "my_variable" and assign the value 10 to it
+my_variable = 10
+
+# print the value of "my_variable"
+print(my_variable) # Output: 10
+```
+
+In this example we initialised the variable - created a variable named `my_variable` and assigned it the value `10`. We then printed the value of `my_variable` using the `print()` function. The output of the program was `10`, which is the value stored in the variable `my_variable`.
+
+## Variable Declaration
+
+In Python, you don't need to explicitly declare a variable before using it. You can create a variable simply by assigning a value to it. Here's an example:
+
+```python
+# create a variable called "x" and assign the value 5 to it
+x = 5
+
+# print the value of "x"
+print(x) # Output: 5
+```
+
+In this example, we created a variable named `x` and assigned it the value `5`. The variable `x` is automatically created when the assignment statement is executed.
+
+However, it is important to note that variables in Python are dynamically typed, which means that the data type of a variable is determined at runtime based on the type of the value assigned to it. This means that the same variable can hold different types of values at different times. For example:
+
+```python
+# create a variable called "y" and assign the value 5 to it
+y = 5
+
+# change the value of "y" to a string
+y = “hello"
+
+# print the value of "y"
+print(y) # Output: hello
+```
+
+In this example, we first created a variable named `y` and assigned it the value `5`. We then changed the value of `y` to a string `hello`. The program printed the value of `y`, which is the string `hello`.
+
+## Data Types
+
```

Comment:
Let's add some sentence here. For example "There are N data types in Ptyhon" and list them (as a HTML list).